srec_info/main.cc

```diff
@@ -37,7 +37,7 @@ main(int argc, char **argv) -> int
 {
     srecord::arglex_tool cmdline(argc, argv);
     cmdline.token_first();
-    typedef std::vector<srecord::input::pointer> infile_t;
+    using infile_t = std::vector<srecord::input::pointer>;
```

Comment:
Doing mostly C++98 and C myself, I always used typedef.
For function pointers (SRecord has only 1) it's more readable as [C++ Core Guidelines](https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#Rt-using) say.

Reply:
I can see that being nice on function pointers and pushing back on the core guidelines is a fools errand!

Are you suggesting that we do this selectively for our one function pointer? Or do we need to do it for all?

Reply:
If there were hundreds of function pointer, it may be worth it. Other than that I'm ok with any solution, preferably consistent.
The question is, what will you do with future PRs that use `using`. Accept inconsistencies or tell them to use typdef. (I don't know if that's realistic, as most people do copy'n'paste).

The benefit of this guideline is, that C++ experts have agreed on it (including Stroustrup). For new code I would simply follow it, but for existing one, it's just a criteria for the decision.

Reply:
That's a really good point and it gets even more challenging depending on whether they base their PR on SRecord code (as is often the case for a new file format) or externally. Consistency is my ultimate preference so let's go with the modern approach. 

Let's go with `using` .